File: src/main/java/io/pageobject/generator/locator/protractor/NgBindLocatorSource.java
package io.pageobject.generator.locator.protractor;

import io.pageobject.generator.GeneratorContext;
import io.pageobject.generator.attribute.HtmlAttributes;
import io.pageobject.generator.locator.AbstractLocatorSource;
import org.jsoup.nodes.Element;

import static io.pageobject.generator.attribute.Attributes.extractAttributeValue;

public class NgBindLocatorSource extends AbstractLocatorSource {

    @Override
    public String extractLocatorValue(Element element, GeneratorContext context) {
        return extractAttributeValue(element, HtmlAttributes.NG_BIND);
    }

    public String locatorTemplate() {
        return "by.exactBinding('%s')";
    }

}

File: lib/locator/protractor/ng-bind-locator-source.js
"use strict";

const AbstractLocatorSource = require('../abstract-locator-source');
const _ = require('lodash');
const extractAttributeValue = require('../../attribute/attributes').extractAttributeValue;
const HtmlAttribute = require('../../attribute/html-attribute');

const template = _.template("by.exactBinding('<%= value %>')");

class NgBindLocatorSource extends AbstractLocatorSource {

  extractLocatorValue(context) {
    return extractAttributeValue(context.element, HtmlAttribute.NG_BIND);
  }

  locatorTemplate() {
    return template;
  }

}

module.exports = NgBindLocatorSource;